### DISCUSSION TITLE
caught segfault address 0x7f868e894000, cause 'memory not mapped'

### DISCUSSION THREAD

**Discussion by showteeth** [brianhie/scanorama#134] [Q&A]
Hi,    
Thanks for providing [Scanorama](https://github.com/brianhie/scanorama) for the integration of single-cell data.
I'm getting an error when applying Scanorama to `panc8` dataset through `reticulate` in R. The code I used:
```R
data.integrated <- scanorama$integrate(data.list, features.list)
```

The error message:
```R
 *** caught segfault ***
address 0x7f868e894000, cause 'memory not mapped'
```

The environment:

- `R`: 4.0.3
- `Seurat`: 4.0.1
- `scanorama`: 1.7.2
- `reticulate`: 1.26

**Comment by showteeth** [comment]
I solved this problem by adding `approx = FALSE` according to [troubleshooting](https://github.com/brianhie/scanorama#troubleshooting). 
  However, when solving this problem, I found another problem. When applying `Scanorama` to `ifnb` dataset:

- I succeed without adding `approx = FALSE`
- and when I added `dimred = as.integer(50)`, I got `address 0x7fb5175ae000, cause 'memory not mapped'`
- similar to the above solution, I added `approx = FALSE`, I got `address 0x7f0d65c71000, cause 'invalid permissions'`

I'm completely confused!

**Reply by brianhie** [reply]
This is quite strange -- I am not sure what is happening. This seems to be caused by some kind of compatibility issue between the packages, probably in how reticulate is treating the numpy/scipy arrays. When I tried integration with R a while back, it seemed to work with R version 3.5.1 and reticulate version 1.10.
